### PR TITLE
modified assn 2

### PR DESCRIPTION
**Pull request by j1george**
modified and improved code. eliminated duplicate code. reordered things to make more sense

### CODE REVIEW

File: ccsf/cs132a_ruby/assn_02/arr2cols.rb [hunk 1]
```diff
@@ -97,38 +97,31 @@
...
-    puts
-end
+all_arrays.each do |array|
+  puts array.last()
 
-puts
+  maxVal = []
+  array.first().each do |row|
+    row.each_with_index do |col, i|
+      maxVal[i] = [maxVal.at(i) || 0, col.to_s.length].max
```
Comment (KellyAH) [inline ccsf/cs132a_ruby/assn_02/arr2cols.rb:112]: There's alotta complex stuff going on in line 112 where your'e setting values in maxVal array. Can you walk me through exactly what are you doing? Are you populating the array with the length of the longest string in the array, so you can use it for setting column width? why do you need this part "[maxVal.at(i) || 0, "

**Comment by KellyAH** [comment]
wow thanks so much for showing me a better way to implement a solution to my ccsf ruby homework.
I'm still wrapping my head around it all.

**Comment by j1george** [comment]
no problem. thanks for putting up your code

File: ccsf/cs132a_ruby/assn_02/arr2cols.rb (hunk 1 above)
Comment (KellyAH) [reply ccsf/cs132a_ruby/assn_02/arr2cols.rb:112]: nevermind. I reverse engineered it and figured it out. Nice job!

row.each_with_index loop is iterating over each value in the array and resolving it's length. Then it populates maxVal array with that length. 

row :["Jim", "bullfrog", 99]
lengths of each value = [3,8,2]

On 1st iteration, maxVal is an empty array. So [maxVal.at(index) returns nil which resolves to false. So the or || statement triggers a return of 0. This auto creates an array of [0, item.to_s.length].  
Then it resolves the highest value via .max method. 

In the case of value 'Jim' , array contains [0,3] and .max method returns 3.

Then maxVal is populated with value, 3 which is set at 0 index position. maxVal = [3]. 

This is repeated on all remaining items in the array, with each new max value set to a different index in maxVal. Until each value in the row has been evaluated.

So after the loop executes on the 1st row :["Jim", "bullfrog", 99], maxVal = [3,8,2].

Now that maxVal is populated with values, the 1st or || condition  'maxVal.at(index)' no longer returns nil. But returns the max length value that resolved in the previously iterated row.

 So after the loop executes on the 2nd row: ["Mack the Knife", "caterpillar", 12]
who's values has lengths: [14,11,2]
and max method returns the largest number.

maxVal is updated to = [14,11,2].

this is how all columns in the escargot_player_data 2 dimensional array is evaluated to find out the length of the longest value. And it's used to find out the optimal column width so data columns are  printed to screen and no data is cut off.

**Review by KellyAH — APPROVED** [review]
Please simplify line 112 'maxVal[i] = [maxVal.at(i) || 0, col.to_s.length].max'
 
I don't see why you need an or '||'

**Comment by j1george** [comment]
poorly named variable on my part. that || could also be avoided using an if check beforehand.

it would result in more code, but it would be a worthy tradeoff considering how much more readable it would be

**Pull request merged**